### DISCUSSION TITLE
Best way to have formKey that needs to be used in a widget down the tree

### DISCUSSION THREAD

**Discussion by mfizz1** [danvick/flutter_form_builder#1277] [Q&A]
As the question states, for example, if you have the Floating action button, and many widgets down the way there is a form (for clean readable code) . What is the best way to pass the key down, without running into duplicate key issue?

**Comment by deandreamatias** [comment]
I think that the best way is creating a MVVM pattern. On your Viewmodel still the form key and then consume on your widgets
